src/security/Guardian.sol

```diff
@@ -0,0 +1,179 @@
+/**
+ * Created by Arcadia Finance
+ * https://www.arcadia.finance
+ *
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+pragma solidity ^0.8.0;
+
+import "../../lib/solmate/src/auth/Owned.sol";
+
+/**
+ * @dev This module provides a mechanism that allows authorized accounts to trigger an emergency stop
+ *
+ */
+abstract contract Guardian is Owned {
+    address public guardian;
+
+    /*
+    //////////////////////////////////////////////////////////////
+                            EVENTS
+    //////////////////////////////////////////////////////////////
+    */
+
+    event GuardianChanged(address indexed oldGuardian, address indexed newGuardian);
+    event PauseUpdate(
+        address account,
+        bool repayPauseUpdate,
+        bool withdrawPauseUpdate,
+        bool borrowPauseUpdate,
+        bool supplyPauseUpdate,
+        bool liquidationPauseUpdate
+    );
+
+    /*
+    //////////////////////////////////////////////////////////////
+                            STORAGE
+    //////////////////////////////////////////////////////////////
+    */
+    bool public repayPaused;
+    bool public withdrawPaused;
+    bool public borrowPaused;
+    bool public depositPaused;
+    bool public liquidationPaused;
+    uint256 public pauseTimestamp;
+
+    constructor() Owned(msg.sender) {
+        repayPaused = false;
+        withdrawPaused = false;
+        borrowPaused = false;
+        depositPaused = false;
```

Comment:
add `liquidationPaused`, although none of the bools are needed here, they're initialized as false